extensions/cornerstone/src/components/WindowLevelActionMenu/WindowLevelActionMenu.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { ReactElement, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import classNames from 'classnames';
+import { AllInOneMenu, useViewportGrid } from '@ohif/ui';
+
+export type WindowLevelPreset = {
+  description: string;
+  window: string;
+  level: string;
+};
+
+export type WindowLevelActionMenuProps = {
+  viewportId: string;
+  viewportElem: HTMLElement;
+  presets: Record<string, Array<WindowLevelPreset>>;
+  onSetWindowLevel: (props) => void;
+  verticalDirection: AllInOneMenu.VerticalDirection;
+  horizontalDirection: AllInOneMenu.HorizontalDirection;
+};
+
+export function WindowLevelActionMenu({
+  viewportId,
+  viewportElem,
+  presets,
+  onSetWindowLevel,
+  verticalDirection,
+  horizontalDirection,
+}: WindowLevelActionMenuProps): ReactElement {
+  const { t } = useTranslation('WindowLevelActionMenu');
+
+  const [viewportGrid] = useViewportGrid();
+  const { activeViewportId } = viewportGrid;
+
+  const [vpHeight, setVpHeight] = useState(viewportElem?.clientHeight);
```

Comment:
There should be a useEffect also that depend on the viewportElem

useEffect(() => {
  if (viewportElem) {
    setVpHeight(viewportElem.clientHeight);
  }
}, [viewportElem]);